Core/Core/Common/CommonUI/CoreWebView/View/CoreWebView.swift

```diff
@@ -400,6 +403,57 @@ open class CoreWebView: WKWebView {
         guard contentHeight.isFinite else { return }
         sizeDelegate?.coreWebView(self, didChangeContentHeight: contentHeight)
     }
+
+    public func resetEnvironment(_ env: AppEnvironment, completion: (() -> Void)? = nil) {
+        self.env = env
+        self.copyRootDomainCookies(for: env, completion: completion)
+    }
+
+    private func copyRootDomainCookies(for env: AppEnvironment, completion: (() -> Void)?) {
+        guard env.isRoot == false else {
+            completion?()
+            return
+        }
+
+        guard
+            let rootDomain = AppEnvironment.shared.api.baseURL.host(),
+            let newDomain = env.api.baseURL.host()
+        else {
+            completion?()
+            return
+        }
+
+        configuration
+            .websiteDataStore
+            .httpCookieStore
+            .getAllCookies()
+            .receive(on: DispatchQueue.main)
```

Comment:
If we expect the `completion` callback on the main thread then we should add this to after the `flatMap` because that can replace the used thread. I'd guess cookies can be used from background threads so I'm not sure if we need to process them on the main queue. Since this is a quite complex method affecting a core component, let's add some unit tests to it.